testing/README.md

```diff
@@ -0,0 +1,35 @@
+# Helpful settings for testing
+
+The plugin will run if you have configured kubectl locally (and have the permissions required to watch/list/read secrets).  
+For full integration testing then this directory contains some files useful for *developer* testing the plugin inside a Jenkins running inside a kubernetes (tested with GKE).
+
+They may require some small tweaks for your environment (as it will be different to mine), but if so please don't attempt to commit them back :-)
+
+## Testing
+
+### Initial setup...
+
+1. Create the testing namespace  `kubectl apply -f testing-namespace.yaml`
+2. create a service user to run Jenkins  `kubectl apply -f service-account.yaml`
+3. Create a role to allow secret reading `kubectl apply -f secret-reader-role.yaml`
+4. Create a role binding to bind the role to the service user. `kubectl apply -f secret-reader-role-binding.yaml`
+
+### Deploying / upgrading Jenkins / plugin
+
+1. Build the plugin (mvn verify)
+2. build the docker image  (`docker build .. -f Dockerfile`)

```

Comment:
If you do a add `-t <tagname>` it will build the image with that tagname rather than a random name so you don't have to do the explicit `docker tag` below.